test/services/iothub/iothubClient-tests.js

```diff
@@ -0,0 +1,292 @@
+// You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+var suite;
+var client;
+var should = require('should');
+var util = require('util');
+var msRestAzure = require('ms-rest-azure');
+var SuiteBase = require('../../framework/suite-base');
+var IoTHubClient = require('../../../lib/services/iothub/lib/iotHubClient');
+var testPrefix = 'iothub-tests';
+var resourceGroupName = 'nodetestrg';
+var resourceName = 'nodeTestHub';
+var location = 'eastus';
+var operationId;
+var subscriptionId;
+var iotHubExists = false;
+var maxRetryCount = 120;
+var retryIntervalInSeconds = 5;
+var url = require('url');
+var consumerGroupName = 'testconsumergroup';
+var eventsEndpointName = 'events';
+
+describe('IoTHub', function () {
+
+    before(function (done) {
+        suite = new SuiteBase(this, testPrefix);
+        suite.setupSuite(function () {
+            client = new IoTHubClient(suite.credentials, suite.subscriptionId, null);
+            subscriptionId = suite.subscriptionId;
+        });
+
+        if (suite.isPlayback) {
+            client.longRunningOperationRetryTimeout = 0;
+        }
+
+        suite.createResourcegroup(resourceGroupName, location, function (err, result) {
+            should.not.exist(err);
+        });
+
+        done();
+    });
+
+    after(function (done) {
+        suite.teardownSuite(done);
+    });
+
+    beforeEach(function (done) {
+        suite.setupTest(done);
+    });
+
+    afterEach(function (done) {
+        suite.baseTeardownTest(done);
+    });
+
+
+    describe('IoTHub Lifecycle Test', function () {
+        it('should check if the iothub name exists', function (done) {
+
+            var operationInputs = {
+                name: resourceName
+            };
+
+            client.iotHubResource.checkNameAvailability(operationInputs, function (err, result, request, response) {
+                should.not.exist(err);
+                if (!result.nameAvailable) {
+                    iotHubExists = true;
+                    console.log('Hub exists');
+                } else {
+                    console.log('Hub does not exist');
+                }
+                done();
+            });
+        });
+
+        it('should delete a hub if it exists successfully', function (done) {
+
+            if (iotHubExists) {
+                client.iotHubResource.deleteMethod(resourceGroupName, resourceName, null, function (err, result, request, response) {
+                    should.not.exist(err);
+                    done();
+                });
+            } else {
+                console.log('Hub does not exist. Skipping Delete');
```

Comment:
please remove all the console.log() statements. They pollute the test run output on console.
Apart from checking that the error does not exist, it would be nice if you do assertions on the received result in required scenarios.